### PR TITLE
Custom groups and timer fix matrix

### LINKED ISSUE #209: Improve channel groups selection
This plugin allows only to pick among three options: All groups, One group, Favourites group.
I think it would be a very good improvement (and not so difficult to be coded) to allow the choice of a list of channel groups (i.e. a comma separated list or strings written in a text file).

'All groups' option is often the best one, but with a lot of channels, system could become very slow,
in particular if a lot of iptv channels have been added in Enigma's services.

### PR DESCRIPTION
**Pull request by phunkyfish**
v4.2.0
- Added: Custom Channel Groups, closes #209
- Added: Connection manager improvements
- Added: Support hidden entries for backend channel numbers
- Fixed: Timer descprition for providers who only use long descrption

### CODE REVIEW

**Comment by phunkyfish** [comment]
@ksooo may look at a suggestion from @hudokkow on how to support multiple channel groups with out an xml file. So will ping back once I've tried it out.

**Comment by phunkyfish** [comment]
@ksooo this is ready when you are.

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po [hunk 1]
```diff
@@ -832,7 +832,7 @@ msgstr ""
 
 #help: Connection - connectionchecktimeout
 msgctxt "#30610"
-msgid "The value in seconds to wait for a connection check to complete before failure. Useful for tuning on older Enigma2 devices."
+msgid "The value in seconds to wait for a connection check to complete before failure. Useful for tuning on older Enigma2 devices. Note, this setting should rarely need to be changed. It's more likely the interval below will have the desired effect."
```
Comment (ksooo) [inline pvr.vuplus/resources/language/resource.language.en_gb/strings.po]: I would always be very careful with statements like 'below' for describing the relation between UI elements. Who guarantees that the settings will be rendered by the UI top-down as specified in settings xml file? And if somebody changes the order of settings in the xml file later, I bet they forget to change this string.

**Review by ksooo — COMMENTED** [review]
Not sure I like the hard coded number of channel groups (code looks really very ugly to me), but if it works... your decision.

File: pvr.vuplus/resources/language/resource.language.en_gb/strings.po (hunk 1 above)
Comment (phunkyfish) [reply pvr.vuplus/resources/language/resource.language.en_gb/strings.po]: Ah, missed that thanks

**Comment by phunkyfish** [comment]
I’m not a fan of the hard coded number either but stumped as to how else to do this so users would not have to leave the UI.

Maybe they’ll introduce a dynamic string list control at some point 😉

**Comment by ksooo** [comment]
> Maybe they’ll introduce a dynamic string list control at some point

 Ah, we have that in Kodi, but it seems this functionality is not available for addons.

**Review by ksooo — APPROVED** [review]
Thanks, man.

**Comment by phunkyfish** [comment]
> Ah, we have that in Kodi, but it seems this functionality is not available for addons.
 
Pity, maybe it will make its way over at some point.

**Pull request merged**